### ISSUE TITLE
TraceView should display a reference to the previous and next Traces in a Frontend App

### ISSUE THREAD

**Issue by thinkocapo** [getsentry/sentry#72929]
### Problem Statement

- You can already see them backwards+forwards in the SessionReplay.Traces tab :heavy_check_mark:, so why not see it this way from the Trace itself? 
- Why have to go through the breadcrumbs, which are only backwards-looking (i.e. you don’t have breadcrumbs for what happens in the future, only the past). You have to open the trace,then scroll down to breadcrumbs to find them.
- TracesExplorer shows a bunch of traces but hard to visualize them occurring across a user journey. You can’t re-order them nor can you specify an order on the x-axis in a Dashboard widget
- Traces are backwards-looking: you can reference via breadcrumbs to a previous transaction/trace but not to a future one.
- Frontend use case only
![image](https://github.com/getsentry/sentry/assets/8920574/2b8a27fb-11da-4383-9e14-4a909d734bde)

### Solution Brainstorm

In the Trace View, don't only show the pageload for /product-join but insert the trace (was it /homepage? /login?) in the row above it.
![image](https://github.com/getsentry/sentry/assets/8920574/8560184c-740f-49b9-b368-3837004f89dc)

Session Replays already do this. Can TraceExplorer + TraceView do it too?

### Product Area

Performance

**Comment by JonasBa** [comment]
@thinkocapo We would like to find a way to do this for all traces, not just ones capture during replay. The issue is that today we do not have any glue between these traces (in the case of replay, each trace has a replay context id which acts as the glue). We'd ideally want to extend such a mechanism to all SDKs, though I'm not sure how doable that would be. 

The plan right now is to add an experimental session tag to the traces in sentry, and evaluate how well it works and how feasible it is to possibly introduce as an SDK integration

cc @Dhrumil-Sentry

**Comment by Lms24** [comment]
closing as a duplicate of https://github.com/getsentry/projects/issues/267 (which is a tracking issue for the entire trace linking project). Once this is done, we will have exactly what's being asked here: previous and next trace buttons in the trace view. (fwiw, this is already available in the `sentry` org for internal testing but can only GA once the trace view is based on EAP).
